Example/Messaging/Sample/Sample/AppDelegate.swift

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import CoreData
+import FirebaseCore
+import FirebaseMessaging
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate {
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    FirebaseApp.configure()
+    
+    let center = UNUserNotificationCenter.current()
+    center.requestAuthorization(options: [.alert, .sound, .badge]) { (granted, error) in
+      if error != nil {
+        print("Failed requesting notification permission: ", error ?? "")
+      }
+    }
+    application.registerForRemoteNotifications()
+    return true
+  }
+
+
+  // MARK: UISceneSession Lifecycle
+
+  func application(_ application: UIApplication, configurationForConnecting connectingSceneSession: UISceneSession, options: UIScene.ConnectionOptions) -> UISceneConfiguration {
+    // Called when a new scene session is being created.
+    // Use this method to select a configuration to create the new scene with.
+    return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
+  }
+
+  // MARK: - Core Data stack
```

Comment:
nit: is the Core Data stack necessary?

Reply:
That's just boilerplate generated by Xcode if you check the "Use Core Data" box. If there isn't any data being saved to Core Data, the save calls aren't doing anything and can also be removed.

Reply:
Yes, but they are called in sceneDelegate and can't be simply removed. persistentContainer is used to create contentView in sceneDelegate.

Reply:
I can remove saveContext tho.

Reply:
SGTM, we can revisit this if it's found to be unnecessary in the future.

Reply:
Done. I wonder what the use case this is. For messaging to update the data object identity, we need to implement MessagingDelegate in SceneDelegate. I wonder with this context, we can just do it in appDelegate.